MoreLinq.Test/RandomSubsetTest.cs

```diff
@@ -206,6 +206,25 @@ public void TestRandomSubsetIsIdempotent()
             Assert.That(sequence, Is.EqualTo(sequenceClone));
         }
 
+        /// <summary>
+        /// Verify that RandomSubset produces subset where all elements belongs to original sequence.
+        /// </summary>
+        [Test]
+        public void TestRandomSubsetReturnsSameElementsThatSequence()
+        {
+            const int count = 100;
+            var sequence = Enumerable.Range(1, count);
+            var resultA = sequence.RandomSubset(count);
```

Comment:
It should be rare, but possible for RandomSubset to return the same source sequence. Then this test would occasionally fail, without any change in the sources or the underlying libraries

Reply:
A simple workaround would be to send a custom `Random` implementation as the second argument, one that will never allow for this to happen.

Reply:
I thought that, but since source has 100 elements, the probability of return the same source sequence is 1 in 100 factorial.

Without this validation, one can change implementation to always return source (what is obvious wrong) and all tests will pass.

Reply:
Note that `resultB` does not suffer from this problem, so I figure that you could just keep that one?

Reply:
I removed resultA from test. (y)